src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -73099,6 +73099,307 @@ static void Main()
+            var iDefinition = comp.GetMember<NamedTypeSymbol>("C.I");
+            Assert.Equal("C<T>.I", iDefinition.ToTestDisplayString());
+            Assert.True(iDefinition.IsDefinition);
+
+            var cDefinition = iDefinition.ContainingType;
+            Assert.Equal("C<T>", cDefinition.ToTestDisplayString());
+            Assert.True(cDefinition.IsDefinition);
+
+            var c2 = cDefinition.Construct(ImmutableArray.Create(TypeWithAnnotations.Create(cDefinition.TypeParameters.Single(), NullableAnnotation.NotAnnotated)));
+            Assert.Equal("C<T>", c2.ToTestDisplayString());
+            Assert.False(c2.IsDefinition);
+
+            AssertHashCodesMatch(cDefinition, c2);
+
+            var i2 = c2.GetTypeMember("I");
+            Assert.Equal("C<T>.I", i2.ToTestDisplayString());
+            Assert.False(i2.IsDefinition);
+
+            AssertHashCodesMatch(iDefinition, i2);
+
+            var c3 = cDefinition.Construct(ImmutableArray.Create(TypeWithAnnotations.Create(cDefinition.TypeParameters.Single(), NullableAnnotation.Annotated)));
+            Assert.Equal("C<T?>", c3.ToTestDisplayString());
+            Assert.False(c3.IsDefinition);
+
+            AssertHashCodesMatch(cDefinition, c3);
+
+            var i3 = c3.GetTypeMember("I");
+            Assert.Equal("C<T?>.I", i3.ToTestDisplayString());
+            Assert.False(i3.IsDefinition);
+
+            AssertHashCodesMatch(iDefinition, i3);
+        }
+
+        [Fact]
+        [WorkItem(30673, "https://github.com/dotnet/roslyn/issues/30673")]
+        public void TypeSymbolGetHashCode_ContainingType_Generic()
+        {
+            var text = @"
+class C<T> where T : class
+{
+    interface I<U> where U : class { }
+}
+";
+
+            var comp = CreateNullableCompilation(text);
+            var iDefinition = comp.GetMember<NamedTypeSymbol>("C.I");
+            Assert.Equal("C<T>.I<U>", iDefinition.ToTestDisplayString());
+            Assert.True(iDefinition.IsDefinition);
+
+            var cDefinition = iDefinition.ContainingType;
+            Assert.Equal("C<T>", cDefinition.ToTestDisplayString());
+            Assert.True(cDefinition.IsDefinition);
+
+            var c2 = cDefinition.Construct(ImmutableArray.Create(TypeWithAnnotations.Create(cDefinition.TypeParameters.Single(), NullableAnnotation.NotAnnotated)));
+            var i2 = c2.GetTypeMember("I");
+            Assert.Equal("C<T>.I<U>", i2.ToTestDisplayString());
+            Assert.True((object)i2.OriginalDefinition == iDefinition);
+            AssertHashCodesMatch(i2, iDefinition);
+
+            var i2constructed = i2.Construct(iDefinition.TypeParameters.Single());
+            Assert.Equal("C<T>.I<U>", i2constructed.ToTestDisplayString());
+            AssertHashCodesMatch(iDefinition, i2constructed);
+
+            var i2b = i2.Construct(ImmutableArray.Create(TypeWithAnnotations.Create(i2.TypeParameters.Single(), NullableAnnotation.Annotated)));
```

Comment:
>var i2b = i2.Construct(ImmutableArray.Create(TypeWithAnnotations.Create(i2.TypeParameters.Single(), NullableAnnotation.Annotated))); [](start = 12, length = 132)

Consider also adding scenario when ```iDefinition``` is constructed this way. Consider also testing scenario when ```iDefinition.TypeParameters``` are used to construct ```i2``` in the same manner.

Reply:
I'll stop by tomorrow to understand better.

---
In reply to: [283579218](https://github.com/dotnet/roslyn/pull/35613#discussion_r283579218) [](ancestors = 283579218)